### PR TITLE
Add reference_attn & reference_adain support for sdxl

### PR DESCRIPTION
**Pull request by zideliu**
# What does this PR do?
It provides reference support for sdxl model, based on [stable_diffusion_reference](https://github.com/huggingface/diffusers/tree/main/examples/community#stable-diffusion-reference)

Fixes # (issue)

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: examples/community/README.md [hunk 1]
```diff
@@ -1850,3 +1850,38 @@ for obj in range(bs):
...
+
+
+```py
+import torch
+from PIL import Image
+from diffusers.utils import load_image
+
+input_image = load_image("https://hf.co/datasets/huggingface/documentation-images/resolve/main/diffusers/input_image_vermeer.png")
+
+pipe = StableDiffusionXLReferencePipeline.from_pretrained(
```
Comment (DN6) [inline examples/community/README.md:1865]:
```suggestion
pipe = DiffusionPipeline.from_pretrained(
    "stabilityai/stable-diffusion-xl-base-1.0",
    custom_pipeline="stable_diffusion_xl_reference",
    torch_dtype=torch.float16,
    use_safetensors=True,
    variant="fp16").to('cuda:0')
```

File: examples/community/README.md [hunk 2]
```diff
@@ -1850,3 +1850,38 @@ for obj in range(bs):
...
+### Stable Diffusion XL Reference
+
+This pipeline uses the Reference . Refer to the [stable_diffusion_reference](https://github.com/huggingface/diffusers/blob/main/examples/community/README.md#stable-diffusion-reference).
+
+
+```py
+import torch
+from PIL import Image
+from diffusers.utils import load_image
+
```
Comment (DN6) [inline examples/community/README.md:1862]:
```suggestion
from diffusers import DiffusionPipeline
from diffusers.schedulers import UniPCMultistepScheduler
```

**Comment by DN6** [comment]
@zideliu PR looks good. Could you provide some example generations from the pipelines to add to the README so that other users can see the expected types of output from the pipeline.

**Comment by zideliu** [comment]
> @zideliu PR looks good. Could you provide some example generations from the pipelines to add to the README so that other users can see the expected types of output from the pipeline.

OK!

**Comment by DN6** [comment]
Thanks for adding the examples @zideliu. I took a look at them, but I can't seem to see any difference between the output image and reference image. Is it possible to try creating an output image where the difference is more obvious?

**Comment by zideliu** [comment]
> Thanks for adding the examples @zideliu. I took a look at them, but I can't seem to see any difference between the output image and reference image. Is it possible to try creating an output image where the difference is more obvious?

OK !

**Review by DN6 — APPROVED** [review]
LGTM 👍🏽

**Comment by DN6** [comment]
@zideliu Nicely done.

**Comment by zideliu** [comment]
OK ~

**Comment by DN6** [comment]
Nice job @zideliu 👍🏽

**Comment by blx0102** [comment]
I tried some images, but the result somehow not really good, the face is usually distorted. Is this normal?
screenshot-20230828-152256

**Comment by blx0102** [comment]
@zideliu Have tried but it didn't help

**Pull request merged**